packages/web-components/src/components/va-file-input/va-file-input.tsx

```diff
@@ -122,6 +158,44 @@ export class VaFileInput {
     return this.buttonText ? this.buttonText : 'Upload file';
   };
 
+  private formatFileSize = (filesSize): string => {
```

Comment:
I'm not sure I understand what's happening with this function or why it's needed. Do you mind adding clarifying it here in the PR or as a comment in the file?

Reply:
Adding the following comment:
`/**
   * Converts the size of a file from bytes to a more human-readable format for
   * rendering the file size label. This function calculates the file size in
   * appropriate units (B, KB, MB, GB, TB) based on the size provided. It uses
   * logarithmic scaling to determine the unit, then formats the size to one
   * decimal place for units KB and above.
   *
   * @param {number} filesSize - The size of the file in bytes.
   * @returns {string} The file size formatted as a string with the appropriate unit.
   */
`